### PR TITLE
rewrite docs

### PR DESCRIPTION
**Pull request by nslater**

### CODE REVIEW

File: docs/types.txt [hunk 1]
```diff
@@ -2,81 +2,142 @@
...
-the underlying java array (it is usually safe to cast it to ``Object[]``) if you prefer.
-Otherwise, to have JDBC handle the type mapping stuff for you, you can use ``.getResultSet()``
-and use the related ``ResultSetMetaData`` or another way to get the array values
-converted to Java POJOs. The ``ResultSet`` will have one column with the inner array type
-and the name of the array field (``Array.getBaseType()``) and as much rows as
-there are elements in the array.
+FIXME: can an engineer please add explanatory comments to this block of code.
+please explain what is being done here. please also format the code in the most
+idiomatic yet reasonable way (I have given this a go myself) - please also add
+empty lines where it makes most sense
```
Comment (nslater) [inline docs/types.txt]: @chaudum can you check this when you review please

File: docs/types.txt [hunk 2]
```diff
@@ -2,81 +2,142 @@
...
+When inserting arrays using a prepared statement, you must convert your array
+to a `java.sql.Array`_ by the use of ``createArrayOf()``. This function takes
+as its first argument, a CrateDB type as described above and as its second the
+array you want to convert.
+
+You can then use ``setArray()`` to set this converted array.
+
+FIXME: can an engineer please add explanatory comments to this block of code.
+please explain what is being done here. please also format the code in the most
+idiomatic yet reasonable way (I have given this a go myself)
```
Comment (nslater) [inline docs/types.txt]: @chaudum here too

File: docs/types.txt [hunk 3]
```diff
@@ -2,81 +2,142 @@
...
+
+.. CAUTION::
+
+   Objects can be ``null``.
+
+FIXIME: is the above right? which object? the resulting java object?
+
+FIXME: can an engineer please add explanatory comments to this block of code.
+please explain what is being done here. please also format the code in the most
+idiomatic yet reasonable way (I have given this a go myself)
```
Comment (nslater) [inline docs/types.txt]: @chaudum here too

File: README.rst [hunk 4]
```diff
@@ -8,24 +8,28 @@ CrateDB JDBC Driver
...
+This is a `type 4 JDBC driver`_. The driver is written in pure Java, and
+communicates with the database using the `PostgreSQL Wire Protocol`_.
 
 Prerequisites
 =============
 
-The CrateDB JDBC driver 2.x requires a CrateDB version greater than or equal to
-0.56.
+The CrateDB JDBC driver requires `Java 8`_, preferably update 20 or later. We
+recommend using `Oracle’s Java`_ on macOS and `OpenJDK`_ on Linux Systems.
```
[the remark this answers is not in the document]
Comment (chaudum) [reply README.rst]: Correct

File: docs/compatibility.txt [hunk 5]
```diff
@@ -8,54 +9,89 @@ Compatibility
...
+|                     |                 |                                      |
+|                     |                 | Also, support for the experimental   |
+|                     |                 | ``showsubcolumns`` parameter has     |
+|                     |                 | been dropped.                        |
++---------------------+-----------------+--------------------------------------+
+| 2.x                 | >= 2.1.x        | Client needs to connect with a valid |
+|                     |                 | database user to access CrateDB.     |
+|                     |                 |                                      |
+|                     |                 | The default CrateDB user is          |
+|                     |                 | ``cratedb`` and has no password is   |
```
Comment (chaudum) [inline docs/compatibility.txt]: default user is `crate`

**Pull request merged**